### ISSUE TITLE
LBF001.pdf

### ISSUE THREAD

**Issue by bountonw** [bountonw/translate#268]
The pdf looks nice! There were no word wrapping issues. (There aren't many Bible words so far. I added "Jesus" to the forbidden break list in case it is an issue later.) 

- [x] There have been a couple of minor updates to the source docs that have been merged over in PR #267 
- [x] Duang finds the chapter titles to be too big. The size gap between the title and the regular text feels "in your face" and she feels like something more subtle is needed, especially since the content of the book is spiritual, she doesn't want people to subconsciously put up a defense.  
- [x] LBF 13.1 (Chapter 4.1) bold text was changed to italics in the source file
- [x] #269
- [x] Size of Bible codes. In the foot notes, the Bible codes `KJV` `TH1971` `TNCV` etc. are the same height as the surrounding Thai text, however in the body, the font face and size appear different. Can we keep them the same font face and size as the numbers?
- [x] Chapter 4 has two foot notes, both are numbered `1` is this by design so that the numbering starts fresh at each page, or should the numbering reset at each chapter?
- [x] Page 11 of the pdf There is one place  where the spacing needs to be increased. It is marked on the attached pdf.
- [x] The generated pdf has drop down menus for the scriptures that appear to be time stamps. For example 2 Corinthians 5:17. If I hover the mouse over 5:17 I have an option to open the calendar and make an appointment.
- [x] I kept the bold in chapter 3 as the English is ALL CAPS and not italics. 
- [x] #272
- [x] The title needs updating as per the book title found in the `metadata.yaml` file.

That is probably enough for now. 

[LBF001_suggestions.pdf](https://github.com/bountonw/translate/files/15157481/LBF001_suggestions.pdf)

**Comment by kenton-r** [comment]
chapter titles to be too big
This is interesting. Here in Cambodia openness to the wide variety of design in not common. Culturally the people in Cambodia expect a title to be very bold and big. So changing it to something different for the Thai is good.

Size of Bible codes
There are several lines in the blockquotes that are wrapped to two lines because of the Bible version reference. Therefore I reduced the size of it because it is not the main information or a requirement to read. I would prefer to leave them small.

Yes, foot notes are set to start on each new page, but I can change it to start on each chapter.

There are no generated bookmarks in this PDF. What you are seeing is probably software recognized assistance. There is not way for us to change that.

I have not worked on the forward, but I will be looking at the MB copyright page mainly.

Thank you for pointing me to where the Title of the book is located.

**Comment by kenton-r** [comment]
[LbF_Book_1b.pdf](https://github.com/bountonw/translate/files/15172775/LbF_Book_1b.pdf)
This PDF has the updated ICML files and the following changes:

- Chapter titles smaller (weight and size)
- Footnotes cycle with chapters
- Better control of spaces shrinking too small
- Book Title added

Somewhere in the processing we lost the "1 Peter" and "2 Peter" forbidden break list.
Also we probably do not need "พระ" added to the forbidden break because InDesign already recognizes it as a cluster.

**Comment by kenton-r** [comment]
[LbF_Book_1c.pdf](https://github.com/bountonw/translate/files/15172898/LbF_Book_1c.pdf)
InDesign does not refresh footnote styles when the ICML file is updated. So I added it to GREP styles.

**Comment by bountonw** [comment]
@mattleff @kenton-r 

Review of `1c`. Thank you for the quick turn over! I did not read the file this time, but looked at the line endings and the previous issues.

* Chapter titles are much improved! I'll run it by Duang later. But it looks good to me.
* I didn't notice the difference in spacing on the line that I circled earlier. In fact, I printed both the previous and current pdfs, folded the paper, and the space width for that location appears exactly the same as before. This is line `11.3.2`. I have circled it in light purple for reference.

Using pdf reference numbers  `page.paragraph.line`

- [ ] `7.1.2` and `7.1.6` are examples of what I mentioned earlier where the book reference is different from the surrounding text.
- [ ] `8.2.2/3` 1 เธสะโลนกา is split over two lines. Isn't this in our forbidden break file? Or are spaces being ignored there. 
- [ ] BTW copying Thessalonians from the pdf gives this `เธิ์สะโลน์กา` which is really weird. Is something going on behind the scenes that will come back to bite us in the end?
- [ ] 11.2.4/5 อัคร~ทูต needs hyphenated. Which opens up the whole can of worms regarding hyphenation. I have a list of a dozen words to add to the hyphenation dictionary. (They aren't flagged now, but they will be before we are done and definitely will be needed in PP.
- [ ] 12.0.4 Does the placement of the second footnote marker look funny?

[LbF_Book_1c_edit.pdf](https://github.com/bountonw/translate/files/15176282/LbF_Book_1c_edit.pdf)

I should have named it ...suggestions..., but oh well.

**Comment by mattleff** [comment]
> * `8.2.2/3` 1 เธสะโลนกา is split over two lines. Isn't this in our forbidden break file? Or are spaces being ignored there.

This should be fixed with new ICMLs after #270 was merged.

**Comment by bountonw** [comment]
I forgot to highlight it and mention it:

- [ ] `8.1.1/2` พระ~เยซู should have a hyphen and bring the พระ onto the previous line. That line is stretched too far.

**Comment by kenton-r** [comment]
BTW copying Thessalonians from the pdf gives this เธิ์สะโลน์กา which is really weird. Is something going on behind the scenes that will come back to bite us in the end?

It looks like there are a lot of problems with generating PDFs from Asian text (not just Thai).

I have tried several different pieces of software and of generating PDFs.

Native Word files generated with Words builtin PDF generator work good.

PDF generated with Adobe Distiller also work good.

There most be something with Postscript (a part of distiller) that is able to keep all of the characters in the right order.

[Test_copying.md](https://github.com/bountonw/translate/files/15185460/Test_copying.md)

**Comment by kenton-r** [comment]
> Does the placement of the second footnote marker look funny?

![Screenshot 2024-05-02 151411](https://github.com/bountonw/translate/assets/26011584/3772b62d-04dc-4f0e-8423-b718f0ad1aa4)

The number is just a superscript. Is it too big?

**Comment by bountonw** [comment]
(We are still packing)

Perhaps half of the white space added when the chapter after the space is ใ or ไ. (Reserving judgement for โ.)

**Comment by kenton-r** [comment]
Here is an updated PDF.
[LbF_Book_1d.pdf](https://github.com/bountonw/translate/files/15186425/LbF_Book_1d.pdf)

It has the 3 dots with half spaces that Matthew implemented.
The fixed Bible book numbers.

It has the fixed copy and pasting issue -- using Adobe Distiller to create the PDFs.

Fixed space with ใ and ไ.

![fixed](https://github.com/bountonw/translate/assets/26011584/290c913b-ea6d-4f77-b89b-d738c5d4c9e4)

As for the scripture version references, I prefer the smaller text because this not the important part of the scripture reference and is distracting at full size. Here are some examples:

![verse_reference_fullsize](https://github.com/bountonw/translate/assets/26011584/6b5373c4-2ac5-4d43-95fd-399660b11e0f)

![verse_reference_fullsize_condensed](https://github.com/bountonw/translate/assets/26011584/eb6b1fd1-b9a4-41d0-85b6-d7a34eca911a)

![verse_reference_small_bold_grey_condensed](https://github.com/bountonw/translate/assets/26011584/e0ba7736-22f2-4ab8-8822-37e99de6ccaa)

![verse_reference_small_condensed](https://github.com/bountonw/translate/assets/26011584/0af05b72-5070-47dc-ac9d-4cd6beac6fe9)

Manual hyphenation has not been done.

> The generated pdf has drop down menus for the scriptures that appear to be time stamps. 

This is not an issue we can fix because it is generated in the PDF reader that it is trying to assist you with office related tasks.

**Comment by bountonw** [comment]
Am I the only one who finds the compacted scripture references distracting? Not only are they smaller than the surrounding text, they are scroonched (a technical term). So we are doing two things here to differentiate them from the surrounding text. Some of the references later in the book are "translated from the English KJV". In that case will only the KJV be a different size with different kerning? (I think the kerning and possibly font face (it if it different) is what I find so distracting.

**Comment by bountonw** [comment]
@mattleff Is the linting for ellipses complete? I know that you and Kenton implemented some improvements to how ellipses are handled. Can I mark this as complete or are there still issues that need to be dealt with in regards to ellipses?

@kenton-r I have glanced at the new pdf. Nice! I like how the ellipses look and the issue with spacing seems improved. I have not read through word for word to look for any new issues. Until a final decision is made on the font face, kerning, and font size of the scripture references, I will wait to do the detailed read as changes there will effect the layout. I commented above with some of my thoughts on the references. In footnote number 2 we have a very nice size and English font that blends nicely with the surrounding text. 

Regarding footnote `#2` I was thinking maybe a nano or pico space to give it some breathing room.

- [ ] 6.3.4/5 องค์-พระคริสต์ needs hyphenated (I know we are not dealing with hyphens, but I may forget. This is more for my reference to check back later before signing off on the chapters.)

We can't really deal with the hyphens until we are settled with the scripture references as the layout will change and the words needing manually hyphenated will change. 

(I am scanning briefly line by line without reading.)

- [ ] 4.1.7/8 พระ-คริสต์ is a possible hyphenation if that allow something ผู้ที่ to go up to line 6. Line 6 is bloated. (Line 7 looks bloated as well for that matter.)
- [ ] 7.2 whole paragraph. Seems bloated พระ-วจนะ can be hyphenated if that would relieve some of the tension. ปีแล้วปีเล่า on the first line does not need hyphenated, but it is an idiomatic phrase that fits nicely together. The bloatiness may be intentional, though, so as to avoid problems on the the last line มุ่งหมาย should be together. 
- [ ] 8.1.3 Line 1 is very bloated, and line 2 is partially bloated. (I already mentioned hyphenating พระ-เยซู earlier to help solve the bloating on line 1). We can also hyphenate พระ-วจนะ and พระ-เจ้า on line 3. (The TNCV version hyphenates these on nearly every page. The bible is printed on two narrow columns and provides a nice reference on the hyphenation rules.)
- [ ] 9.1.6 The italicized word. In English, all words are surrounded by spaces. However, this is not the case for Thai and italicized words in the middle of a phrase may pose an aesthetic problem. Is there a way of inserting a **tiny** space before and after italicized words that are surrounded by other text (and to not add the space if a real space is already there?
- [ ] 10.1.3 footnote marker. I think there is also an issue of the footnote marker being slightly too low. It is hard to tell as the two examples we have do not occur next to a letter of normal hight.

**Comment by mattleff** [comment]
@bountonw I haven't implemented the linting for ellipsis (#269), but all the ellipsis in LBF are correct, so we should be ok to mark it complete for LBF.

**Comment by kenton-r** [comment]
The footnote inline markers are raised height to the level of top characters and a thin space has been added before it globally.

Here are three different sample of the version references:
Same size as numbers (10pt)
[LbF_ch-04_same-size.pdf](https://github.com/bountonw/translate/files/15235716/LbF_ch-04_same-size.pdf)

Middle size (8pt)
[LbF_ch-04_middle_size.pdf](https://github.com/bountonw/translate/files/15235734/LbF_ch-04_middle_size.pdf)

Small size (7pt and condensed semibold)
[LbF_ch-04_small_size.pdf](https://github.com/bountonw/translate/files/15235741/LbF_ch-04_small_size.pdf)

**Comment by mattleff** [comment]
Rachel and I both prefer the same-size (10pt) version references. The stroke weight of the Thai characters feels heavier than the Latin numerals and characters in all the options, but the 10pt version feels closest. Just an opinion, for what it's worth.

**Comment by bountonw** [comment]
I also like the 10 pt version much better. The TH looks higher than the numbers, but it is the closest thing that matches.

**Comment by kenton-r** [comment]
So looking at fixing some of these issues.

> 11.2.4/5` อัคร~ทูต needs hyphenated. 

> 6.3.4/5 องค์-พระคริสต์ needs hyphenated

These words need to be added to the Forbidden Break word list:
อัครทูต
องค์พระคริสต์

> 8.1.1/2 พระ~เยซู should have a hyphen

This is a strange issue with InDesign wanting to connect this word (Jesus' name) with the next word. 

![Problem_with_breaking_after_Jesus_name_1](https://github.com/bountonw/translate/assets/26011584/602fe975-1e7e-4b88-8fe2-2b70c7bd32c8)

For fixing this, I would like to change the spelling of this word on the Forbidden Break list to only:
พระเย
(InDesign still sees the whole word as a cluster)

> 9.1.6 The italicized word

Can we add this to the processing of the ICML? I can not dynamically added that to the design. So if we can add it to the processing of the Italic styling, I would like to put thin spaces around the italic words. " " This is a thin space.

**Comment by bountonw** [comment]
> So looking at fixing some of these issues.
> 
> > 11.2.4/5` อัคร~ทูต needs hyphenated.
> 
> > 6.3.4/5 องค์-พระคริสต์ needs hyphenated
> 
> These words need to be added to the Forbidden Break word list: อัครทูต องค์พระคริสต์

They need hyphenated. I started to add them to the forbidden break list and then realized that that defeats the purpose.

> > 8.1.1/2 พระ~เยซู should have a hyphen
> 
> This is a strange issue with InDesign wanting to connect this word (Jesus' name) with the next word.
> 
> ![Problem_with_breaking_after_Jesus_name_1](https://private-user-images.githubusercontent.com/26011584/329843532-602fe975-1e7e-4b88-8fe2-2b70c7bd32c8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU1MTg3ODgsIm5iZiI6MTcxNTUxODQ4OCwicGF0aCI6Ii8yNjAxMTU4NC8zMjk4NDM1MzItNjAyZmU5NzUtMWU3ZS00Yjg4LThmZTItMmI3MGM3YmQzMmM4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTEyVDEyNTQ0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM5M2IzYWQxYjQxZTY3NTlkYjhjNDUyMGZiYjI2ZTgxZGI1ZjZjMTk1ZTRmZDhlYzg4ZWZjZTEyM2UzZjM5YzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.o9QLLtAOfxzO5NjGa6UcYw9-H8TvVk6Qy76_Sg1aqWY)
> 
> For fixing this, I would like to change the spelling of this word on the Forbidden Break list to only: พระเย (InDesign still sees the whole word as a cluster)

พระ and เยซู are listed separately on the forbidden break list as there were times that the program in the past broke them. But พระ-เยซู is a valid hyphenation point.

This is a game stopper. It needs to be hyphenatable. There are dozens of words that can take a hyphen that start with พระ. The narrow column TNCV is very helpful in identifying which words should be hyphenated. Since God, Jesus, and many royal language words begin with พระ, it is very common in our text and we need to be able to hyphenate them.

> > 9.1.6 The italicized word
> 
> Can we add this to the processing of the ICML? I can not dynamically added that to the design. So if we can add it to the processing of the Italic styling, I would like to put thin spaces around the italic words. " " This is a thin space.

This would be something for @mattleff to do. Thin space or very thin space. Not sure the options. You can play around with it.

**Comment by kenton-r** [comment]
>  They need hyphenated. I started to add them to the forbidden break list and then realized that that defeats the purpose.

InDesign is more fluid than the need to hyphenate for 3 characters. Those three characters are better put onto the same line.

With the current place that technology for Thai writing is -- There is no possibility of a functional auto hyphenation list. 

HarfBuzz is the primary engine that brakes words and lines based on its clustering patterns. It is used in InDesign and Latex. It does not communicate with a hyphenation engine such as Hunspell. This leaves most of the words that it breaks with no possibility of auto hyphenation.

Forbidden breaks are the only automation that we have available right now.

>   พระ and เยซู are listed separately on the forbidden break list as there were times that the program in the past broke them. But พระ-เยซู is a valid hyphenation point.

Both of these words are now managed as HarfBuzz clusters. Two separate ones. Again there is no way to automate hyphenation for this word because HarfBuzz already recognizes it as two clusters. It will break it with no communication to the Hunspell hyphenation engine. But three characters are easily moved up or down into a line together.

The issue is that somehow the last character is linking dragging the next word with it when the force "no-break" is applied to the whole word. However a workaround for this strange issue is to have the Forbidden Break style no include the last two characters (main and bottom characters). There are so far no other words showing this issue.

>  This is a game stopper. It needs to be hyphenatable. 

You are asking for something that is not yet developed in the whole world.

"Game Stopper" means you are wanting to go to what? What else is there?

**Comment by kenton-r** [comment]
[LbF_Book_1e.pdf](https://github.com/bountonw/translate/files/15288990/LbF_Book_1e.pdf)

This version has changed several way:

- The font for English text and numbers is now Noto Sans Medium.  This weight matches better and the letter and number heights are the same in the font.
- The justification values have been changes slightly to put more space into the end-sentence spaces than to the "bloating" of the text.
- The start of a copyright/forward page.
- The equal sized Bible version references.
- Manual correction for พระเยซู on page 8.2.1

**Comment by kenton-r** [comment]
[LbF_Book_1f.pdf](https://github.com/bountonw/translate/files/15330225/LbF_Book_1f.pdf)

This is the full book with updated changes:

- beginnings of the table of contents
- applied a work-around for Forbidden Break styles. The following word is also forced to join. It works in English correctly, but I think that it is looking for a space to help it to stop. So I removed the Forbidden Break style from the last two characters.

Problems remaining:

- Chapter 34 has a chapter number of "1"
- Subheads are Header3. The chapter number is also Header3. So we need to make the chapter numbers switched to Header2.
- Which copyright information needs to be included about Bibles used in this book?
- There are some words on the Copyright page that need to be translated.
- The word "Content" needs to be translated also for the TOC.
- The authors of this book are E. J. Waggner and A. T. Jones. It is currently showing Ellen White.

**Comment by kenton-r** [comment]
[LbF_Book_1g.pdf](https://github.com/bountonw/translate/files/15346049/LbF_Book_1g.pdf)

Updated with the following changes:

- Correct Hair spaces around italics
- Correct Chapter number and subhead styles
- Added authors names to cover page
- found and added translation for the TOC title

Still needing assistance with:

- Manually changes the number for chapter 34, but still needs to be changed in source.
- Footnotes have become italicized in the ICML. They need to be default style.
- With Google's help translated some of the words on the copyright page. So they need to be checked. Also for attribution would "Digital text processing" be the right terms for Matthew's role.

Question
How hard would it be to temporarily add the removal the Forbidden Break style from the last two characters of each word? Not to delete the letters from the list because hopefully this bug will be fixed. Currently I am doing a search and replace looking for all the Forbidden Breaks and using this GREP pattern:   (.{1}[^(?!.*ะ)]\>)   This excludes the word "พระ" because this one is better to have it connect with the following words. I change the characters just back to the default style.

**Comment by mattleff** [comment]
@kenton-r I've merged a PR to fix LBF chapter number 34 (#288). 

I'm not sure about the footnote italicization issue. In the ICML it looks like the footnotes are styled as `ParagraphStyle/Footnote > Paragraph`. Is it possible to apply the normal paragraph style to that style?

I'll work on sending a PR to try the truncated forbidden break bug workaround.

**Comment by mattleff** [comment]
@kenton-r For the forbidden word break bug workaround, are we wanting to move the style up two characters or two columns? For example, for the word `จริง` which of the following are we wanting:

1. 
```
<CharacterStyleRange AppliedCharacterStyle="CharacterStyle/Forbidden Break">
  <Content>จ</Content>
</CharacterStyleRange>
<CharacterStyleRange AppliedCharacterStyle="$ID/NormalCharacterStyle">
  <Content>ริง</Content>
</CharacterStyleRange>
```


2. 
```
<CharacterStyleRange AppliedCharacterStyle="CharacterStyle/Forbidden Break">
  <Content>จร</Content>
</CharacterStyleRange>
<CharacterStyleRange AppliedCharacterStyle="$ID/NormalCharacterStyle">
  <Content>ิง</Content>
</CharacterStyleRange>
```

**Comment by kenton-r** [comment]
#2 is what I am currently implementing because of character like "ต์" where there are two characters stacked. If only the very last character is selected it still leave the column as Forbidden break and attached to the next word.

**Comment by kenton-r** [comment]
I figured out what was happening with the Italics on the footnote number. For the book, the italics had somehow been set to the defaults for the text tool.

**Comment by kenton-r** [comment]
This is the PDF for beginning editing. Besides the introductory pages, this is exactly as it renders automatically with the ICML files generated. There are no manual interventions yet.

[LbF_Book_editing_1a.pdf](https://github.com/bountonw/translate/files/15418283/LbF_Book_editing_1a.pdf)

Highlighting and comments are available in Acrobat reader I believe. It you want to do something else that is OK too.

**Comment by bountonw** [comment]
This issue has become bloated and we are working with a new pdf now. I am opening a new issue for further discussion and copying any issues from here that are not resolved to issue #296 (I had to create the issue in order to get the issue number. I am still editing the description.)
